clarifai/client/model.py

```diff
@@ -576,3 +576,179 @@ def __str__(self):
+
+    if response.status.code != status_code_pb2.SUCCESS:
+      raise Exception(response.status)
+
+    results = []
+    fields = [
+        "status",
+        "summary",
+        "id",
+        "eval_info",
+        "ground_truth_dataset",
+    ]
+    for eval_metric in response.eval_metrics:
+      for field in eval_metric.DESCRIPTOR.fields_by_name.keys():
+        if field not in fields:
+          eval_metric.ClearField(field)
+      results.append(eval_metric)
+
+    return results
+
+  def evaluate(self,
+               dataset_id: str,
+               dataset_app_id: str = None,
+               dataset_user_id: str = None,
+               eval_id: str = None,
+               extended_metrics: dict = None,
+               eval_info: dict = None) -> resources_pb2.EvalMetrics:
+    """ Run evaluation
+
+    Args:
+      dataset_id (str): Dataset Id.
+      dataset_app_id (str): App ID for cross app evaluation, leave it as None to use Model App ID. Default is None.
+      dataset_user_id (str): User ID for cross app evaluation, leave it as None to use Model User ID. Default is None.
+      eval_id (str): Specific ID for the evaluation. You must specify this parameter to either overwrite the result with the dataset ID or format your evaluation in an informative manner. If you don't, it will use random ID from system. Default is None.
+      extended_metrics (dict): user custom metrics result. Default is None.
+      eval_info (dict): custom eval info. Default is empty dict.
+
+    Return
+      eval_metrics
+
+    """
+
+    metrics = None
+    if isinstance(extended_metrics, dict):
+      metrics = Struct()
+      metrics.update(extended_metrics)
+      metrics = resources_pb2.ExtendedMetrics(user_metrics=metrics)
+
+    eval_info_params = None
+    if isinstance(eval_info, dict):
+      eval_info_params = Struct()
+      eval_info_params.update(eval_info)
+      eval_info_params = resources_pb2.EvalInfo(params=eval_info_params)
+
+    eval_metric = resources_pb2.EvalMetrics(
+        id=eval_id,
+        model=resources_pb2.Model(
+            id=self.id,
+            app_id=self.auth_helper.app_id,
+            user_id=self.auth_helper.user_id,
+            model_version=resources_pb2.ModelVersion(id=self.model_info.model_version.id),
+        ),
+        extended_metrics=metrics,
+        ground_truth_dataset=resources_pb2.Dataset(
+            id=dataset_id,
+            app_id=dataset_app_id or self.auth_helper.app_id,
+            user_id=dataset_user_id or self.auth_helper.user_id,
+        ),
+        eval_info=eval_info_params,
+    )
+    request = service_pb2.PostEvaluationsRequest(
+        user_app_id=self.user_app_id,
+        eval_metrics=[eval_metric],
+    )
+    response = self._grpc_request(self.STUB.PostEvaluations, request)
+    if response.status.code != status_code_pb2.SUCCESS:
+      raise Exception(response.status)
+
+    return response.eval_metrics
```

Comment:
Is it possible to parse the response proto and make it as a dict with only the metric details.It will be a good user experience. WDYT?

Reply:
In this endpoint wrapper, I aim for users to be able to utilize type hints and parse the output data easily. The output of ListEval includes the evaluation ID, dataset information, summary, and status of evaluations. This information is crucial for further work, such as providing the evaluation ID to retrieve detailed evaluation results or checking if a dataset has been evaluated.
I plan to create some helper functions based on these wrappers to make users' lives easier.

Reply:
Got it 👍